### PR TITLE
chore: support IJ 2024.2

### LINKED ISSUE #889: Support  2024.2
Current Next EAP nightly ( 2024.2 ) is failing due to major changes. 
With RC out, time to get  this fixed. 

## What versions of software are you using?
- Operating System:
- IntelliJ version:
- Extension version:
- Are you targeting a kubernetes or an Openshift cluster ? If possible, provide its version:

### PR DESCRIPTION
**Pull request by sbouchet**
## Was the change discussed in an issue?
fixes #889 #823

### CODE REVIEW

File: .github/workflows/validate_IJ_versions.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        IJ: [ IU-2022.3, IU-2023.1, IU-2023.2, IU-2023.3, IU-2024.1 ]
+        IJ: [ 2022.3, 2023.1, 2023.2, 2023.3, 2024.1 ]
```
Comment (adietish) [inline .github/workflows/validate_IJ_versions.yml:12]: add `2024.2`?

**Comment by sbouchet** [comment]
@martinszuc : is there a way to try out my changes for the integration tests tasks ?  
currently it uses the old build.gradle on main and not the one i modified in this PR

**Comment by martinszuc** [comment]
> @martinszuc : is there a way to try out my changes for the integration tests tasks ?  
> currently it uses the old build.gradle on main and not the one i modified in this PR

@sbouchet The way to do it is create and merge PR on your own fork. After the merge to main the tests are triggered automatically after the Java Ci with Grade run ends (which you can cancel). So you would check the Actions tab on your fork. After that I reset the main branch and repeat.

Not ideal but the reason is limitations because of idea licence.

**Comment by sbouchet** [comment]
> > @martinszuc : is there a way to try out my changes for the integration tests tasks ?
> > currently it uses the old build.gradle on main and not the one i modified in this PR
> 
> @sbouchet The way to do it is create and merge PR on your own fork. After the merge to main the tests are triggered automatically after the Java Ci with Grade run ends (which you can cancel). So you would check the Actions tab on your fork. After that I reset the main branch and repeat.
> 
> Not ideal but the reason is limitations because of idea licence.

ok, i tried it out  anfd i can now have a working UI test, but with tests failures.
you can check the result here : https://github.com/sbouchet/intellij-openshift-connector/actions/runs/10389016704/job/28766038969
however, locally i can't debug because it deletes my IdeaProject Folder where i store my work ( and source code ). what is your setup ?

**Comment by martinszuc** [comment]
> @sbouchet  ok, i tried it out anfd i can now have a working UI test, but with tests failures. you can check the result here : https://github.com/sbouchet/intellij-openshift-connector/actions/runs/10389016704/job/28766038969

I believe this could be idea licence not being present. Have you added the decrypt password to your forks secrets?

>however, locally i can't debug because it deletes my IdeaProject Folder where i store my work ( and source code ). what is your setup ?

I dont use IdeaProjects, I have a separate folder for projects. Which is also why I didnt notice this. I will take a look if it needs to be that way. If not, its my oversight from [commit](https://github.com/redhat-developer/intellij-common-ui-test-library/commit/90dc3b991ad091a3463791c05f3f7acb8090d43f) and ill try to fix

**Comment by sbouchet** [comment]
> > @sbouchet  ok, i tried it out anfd i can now have a working UI test, but with tests failures. you can check the result here : https://github.com/sbouchet/intellij-openshift-connector/actions/runs/10389016704/job/28766038969
> 
> I believe this could be idea licence not being present. Have you added the decrypt password to your forks secrets?
> 
> > however, locally i can't debug because it deletes my IdeaProject Folder where i store my work ( and source code ). what is your setup ?
> 
> I dont use IdeaProjects, I have a separate folder for projects. Which is also why I didnt notice this. I will take a look if it needs to be that way. If not, its my oversight from [commit](https://github.com/redhat-developer/intellij-common-ui-test-library/commit/90dc3b991ad091a3463791c05f3f7acb8090d43f) and ill try to fix

it seems i need 0.4.2-snapshot, with this fix https://github.com/redhat-developer/intellij-common-ui-test-library/commit/7336474ed5f2f006f2dbc3e29d2445a03b5d183b

**Comment by sbouchet** [comment]
@martinszuc : tests fails with 
```
Initialize IdeaRunner and start IDE
      Starting IDE, setting ideaIsStarted to true
      [ 137852]   WARN - logger -         Failed on step: Search many 'JTreeFixture's by '//div[@class='JBViewport']/*' (StepWorker.kt_step)
      [ 137852]   WARN - logger -     Failed on step: Start IntelliJ Idea ('IU-2024.1') listening on port 8580 (StepWorker.kt_step)
```
does it means the ide is correctly started but it does not find the correct UI element ? 
the stacks is 
```
java.lang.reflect.InvocationTargetException
          at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
          at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
          at com.intellij.remoterobot.SearchContext$findAll$1.invoke(SearchContext.kt:78)
          at com.intellij.remoterobot.SearchContext$findAll$1.invoke(SearchContext.kt:76)
          at com.intellij.remoterobot.stepsProcessing.StepWorkerKt.step(StepWorker.kt:23)
          at com.intellij.remoterobot.SearchContext$DefaultImpls.findAll(SearchContext.kt:76)
          at com.intellij.remoterobot.RemoteRobot.findAll(RemoteRobot.kt:32)
          at com.redhat.devtools.intellij.commonuitest.fixtures.dialogs.FlatWelcomeFrame.projectsCount(FlatWelcomeFrame.java:224)
          at com.redhat.devtools.intellij.commonuitest.fixtures.dialogs.FlatWelcomeFrame.clearWorkspace(FlatWelcomeFrame.java:115)
          at com.redhat.devtools.intellij.commonuitest.UITestRunner.lambda$runIde$0(UITestRunner.java:97)
          at com.intellij.remoterobot.stepsProcessing.StepWorkerKt.step(StepWorker.kt:23)
          at com.redhat.devtools.intellij.commonuitest.UITestRunner.runIde(UITestRunner.java:70)
          at org.jboss.tools.intellij.openshift.test.ui.runner.IdeaRunner.startIDE(IdeaRunner.java:42)
          at org.jboss.tools.intellij.openshift.test.ui.junit.TestRunnerExtension.beforeAll(TestRunnerExtension.java:33)
          at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
      Caused by: com.intellij.remoterobot.client.IdeaSideException: Wrapped org.assertj.swing.exception.ComponentLookupException: Unable to find component using matcher org.assertj.swing.core.NameMatcher[name='com.intellij.ui.components.JBList[,0,0,250x456,alignmentX=0.0,alignmentY=0.0,border=ViewBorder{myInsets=com.intellij.util.ui.JBInsets[top=0,left=0,bottom=0,right=0], myBorder=null},flags=33554728,maximumSize=java.awt.Dimension[width=250,height=161],minimumSize=,preferredSize=java.awt.Dimension[width=250,height=161],fixedCellHeight=-1,fixedCellWidth=-1,horizontalScrollIncrement=-1,selectionBackground=IJColorUIResource(color=java.awt.Color[r=47,g=101,b=202], name=List.selectionBackground),selectionForeground=IJColorUIResource(color=java.awt.Color[r=1[87](https://github.com/sbouchet/intellij-openshift-connector/actions/runs/10922868695/job/30318144882#step:6:88),g=187,b=187], name=List.selectionForeground),visibleRowCount=8,layoutOrientation=0]', type=javax.swing.JTree, requireShowing=true].
```

**Comment by sbouchet** [comment]
> @martinszuc : tests fails with
> 
> ```
> Initialize IdeaRunner and start IDE
>       Starting IDE, setting ideaIsStarted to true
>       [ 137852]   WARN - logger -         Failed on step: Search many 'JTreeFixture's by '//div[@class='JBViewport']/*' (StepWorker.kt_step)
>       [ 137852]   WARN - logger -     Failed on step: Start IntelliJ Idea ('IU-2024.1') listening on port 8580 (StepWorker.kt_step)
> ```

oh my ... just saw that i'm trying to run IU-2024.1 where my actual PR is to support 2024.2 maybe that's why there is some errors !

**Review by adietish — APPROVED** [review]
lgtm. I especially like that you are now setting `sinceBuild` in the build.

**Review by adietish — APPROVED** [review]
beside my nitpicking LGTM. Awesome, we're almost done having all plugins/lib compile against 2024.2 and compatible with 2024.3 👍

**Comment by adietish** [comment]
@sbouchet do we know why the ITs still fail?

File: src/test/java/org/jboss/tools/intellij/openshift/ui/sandbox/SandboxRegistrationServerMock.java [hunk 2]
```diff
@@ -19,131 +19,133 @@
...
+                "consoleURL": "",
+                "familyName": "",
+                "givenName": "",
+                "status": {
+                    "ready": false,
+                    "reason": "PendingApproval",
+                    "verificationRequired": true
+                },
+                "username": ""
+            }""";
```
[the remark this answers is not in the document]
Comment (sbouchet) [reply src/test/java/org/jboss/tools/intellij/openshift/ui/sandbox/SandboxRegistrationServerMock.java:43]: support in Java 15 and above :D

**Comment by adietish** [comment]
/override "Cluster Integration UI Tests Report" "Public Integration UI Tests Report" "ci/prow/e2e-openshift"

**Review by adietish — APPROVED** [review]
LGTM! Great!

**Pull request merged**